graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/types/generateDirective.kt

```diff
@@ -29,20 +30,27 @@ import kotlin.reflect.KClass
 import kotlin.reflect.KProperty
 import com.expediagroup.graphql.annotations.GraphQLDirective as GraphQLDirectiveAnnotation
 
-internal fun generateDirectives(generator: SchemaGenerator, element: KAnnotatedElement, parentClass: KClass<*>? = null): List<GraphQLDirective> {
+internal fun generateDirectives(
+    generator: SchemaGenerator,
+    element: KAnnotatedElement,
+    location: Introspection.DirectiveLocation,
+    parentClass: KClass<*>? = null
+): List<GraphQLDirective> {
     val annotations = when {
         element is KProperty<*> && parentClass != null -> element.getPropertyAnnotations(parentClass)
         else -> element.annotations
     }
 
     return annotations
         .mapNotNull { it.getDirectiveInfo() }
+        .filter { it.directiveAnnotation.locations.contains(location) }
```

Comment:
should we fail if someone attempts to add directive on invalid location? guess it might be tricky in case of input/output objects that can be represented by the same class

Reply:
That is what we are doing today. So with this change the directive will just not appear in the schema. If someone wants to catch that at Kotlin compile time they should update the `annotation class` `@Target`